server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt

```diff
@@ -198,29 +199,54 @@ object Chapter {
     data class ChapterChange(
         val isRead: Boolean? = null,
         val isBookmarked: Boolean? = null,
-        val lastPageRead: Int? = null // this probably won't be very useful, but for completion's sake
+        val lastPageRead: Int? = null, // this probably won't be very useful, but for completion's sake
+        val delete: Boolean? = null
     )
 
     @Serializable
-    data class ChapterBatchEditInput(
+    data class MangaChapterBatchEditInput(
         val chapterIds: List<Int>? = null,
         val chapterIndexes: List<Int>? = null,
         val change: ChapterChange?
     )
 
-    fun modifyChapters(input: ChapterBatchEditInput, mangaId: Int) {
+    @Serializable
+    data class ChapterBatchEditInput(
+        val chapterIds: List<Int>? = null,
+        val change: ChapterChange?
+    )
+
+    fun modifyChapters(input: MangaChapterBatchEditInput, mangaId: Int? = null) {
         // Make sure change is defined
         if (input.change == null) return
-        val (isRead, isBookmarked, lastPageRead) = input.change
+        val (isRead, isBookmarked, lastPageRead, delete) = input.change
+
+        // Handle deleting separately
+        if (delete == true) {
+            return deleteChapters(input, mangaId)
```

Comment:
Probably shouldn't early return if there are other tasks to do.

```suggestion
            deleteChapters(input, mangaId)
            if (listOfNotNull(isRead, isBookmarked, lastPageRead).isEmpty()) return
```